bundles/org.openhab.binding.govee/src/main/java/org/openhab/binding/govee/internal/GoveeHandler.java

```diff
@@ -151,114 +179,186 @@ public void dispose() {
     }
 
     @Override
-    public void handleCommand(ChannelUID channelUID, Command command) {
-        try {
-            if (command instanceof RefreshType) {
-                // we are refreshing all channels at once, as we get all information at the same time
-                triggerDeviceStatusRefresh();
-                logger.debug("Triggering Refresh");
-            } else {
-                logger.debug("Channel ID {} type {}", channelUID.getId(), command.getClass());
-                switch (channelUID.getId()) {
-                    case CHANNEL_COLOR:
-                        if (command instanceof HSBType hsbCommand) {
-                            int[] rgb = ColorUtil.hsbToRgb(hsbCommand);
-                            sendColor(new Color(rgb[0], rgb[1], rgb[2]));
-                        } else if (command instanceof PercentType percent) {
-                            sendBrightness(percent.intValue());
-                        } else if (command instanceof OnOffType onOffCommand) {
-                            sendOnOff(onOffCommand);
+    public void handleCommand(ChannelUID channelUID, Command commandParam) {
+        Command command = commandParam;
+
+        logger.debug("handleCommand({}, {})", channelUID, command);
+
+        List<Callable<?>> communicationTasks = new ArrayList<>();
```

Comment:
Since this runs sequentially now, can we use Runnables instead? Would allow getting rid of the useless return calls.

Reply:
That is what I wanted to, and tried to do initially. But I couldn't get it to compile. Something wierd about the lamddas. But I will have another try..

Reply:
@maniac103 the problem was twofold
1. we must use Callable rather than Runnable because the methods in the lambdas can throw exceptions which Runnable does not allow
2. the methods in the lambdas were void, so we had to wrap them with an outside `return`; this is now fixed by changing the methods to return a (dummy) value